GANDLF/logging_config.yaml

```diff
@@ -0,0 +1,39 @@
+version: 1
+formatters:
+  detailed:
+    format: "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+  detailed1:
+    format: "%(asctime)s - %(name)s - %(levelname)s - %(pathname)s:%(lineno)d - %(message)s"
```

Comment:
I have kept them because I don't know which format is more convenient. 
What do you believe is the best format? 
Also, it might be better to use different formats. For example, in stdout, we might not want a lot of information, but in the log file, it is useful to have extra information. 
Any thoughts? @VukW @sarthakpati

Reply:
1. IMO using different formatters (short and long) for CLI and debug logs is a good idea, let's do it
2. (detailed format) For me personally `pathname` looks too excessive - on your screenshot it takes almost all the space. I usually prefer to use `module` (python script file name, without the full path) or `module`+`funcName`. It seems enough for easy understanding where exactly message was written.

Example of how `module`+`funcName` instead of `pathname` look like:
`format: "%(asctime)s - %(name)s - %(levelname)s - %(module)s:%(funcName)s:%(lineno)d -  %(message)s"`
```
2024-07-04 11:49:51,148 - py.warnings - WARNING - warnings:_showwarnmsg:109 -  /mnt/c/Users/vykuk/repos/mlc/GaNDLF/check_logging_levels.py:15: UserWarning: a WARNING!!
  warnings.warn("a WARNING!!")  # debug log and stderr
2024-07-04 11:49:54,164 - my-logger - INFO - check_logging_levels:main:30 -  this is logging.INFO message
```
https://docs.python.org/3/library/logging.html#logrecord-attributes

Reply:
Yes, I agree. I like this format, is concise and smaller. I will change it. 
Thanks for your comments. :P